### PR TITLE
[CI] Re-balance concurrency groups to allow more quota for `large` tests

### PR DESCRIPTION
**Pull request by mwtian**
## Why are these changes needed?
Currently nightly tests are unable to finish in a day because of concurrency group limit on `large` tests. This is an attempt to adjust the limits so buildkite can run / finish more tests. I will observe which tests fall into the `enormous` group and adjust the test resource / concurrency group limits again.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
@mwtian You don't need to observe, you can run some code to figure out which tests are enormous.

Something like
```
tests = config.read_and_validate_release_test_collection("release_tests.yaml")
groups = [(concurrency.get_concurrency_group(test), test['name'], test['frequency']) for test in tests]
[group for group in groups if group[0][0] == 'large']
```

**Review by rkooo567 — APPROVED** [review]
This lgtm. (It was actually the next step we thought if the first pr doesn’t fix the issue). One concern is we don’t take into account of gpu tests usage, but we can see how it works out. Also, Note that we had about 1% of capacity related failures last week. We can use this for the comparison point

**Review by krfricke — CHANGES_REQUESTED** [review]
Can we please add a test to `test_buildkite.py` at `testConcurrencyGroups`?

File: release/ray_release/buildkite/concurrency.py [hunk 1]
```diff
@@ -27,10 +28,11 @@
 gpu_cpu_to_concurrency_groups = [
     Condition(min_gpu=9, max_gpu=-1, min_cpu=0, max_cpu=-1, group="large-gpu"),
     Condition(min_gpu=1, max_gpu=9, min_cpu=0, max_cpu=-128, group="small-gpu"),
-    Condition(min_gpu=0, max_gpu=0, min_cpu=513, max_cpu=-1, group="large"),
+    Condition(min_gpu=0, max_gpu=0, min_cpu=1025, max_cpu=-1, group="enormous"),
+    Condition(min_gpu=0, max_gpu=0, min_cpu=513, max_cpu=1024, group="large"),
     Condition(min_gpu=0, max_gpu=0, min_cpu=129, max_cpu=512, group="medium"),
+    Condition(min_gpu=0, max_gpu=0, min_cpu=33, max_cpu=128, group="small"),
     Condition(min_gpu=0, max_gpu=0, min_cpu=0, max_cpu=32, group="tiny"),
-    Condition(min_gpu=0, max_gpu=0, min_cpu=0, max_cpu=128, group="small"),
```
Comment (krfricke) [inline release/ray_release/buildkite/concurrency.py:33]: This should actually stay here so "small" is the fallback (and not tiny)

Comment (krfricke) [reply release/ray_release/buildkite/concurrency.py:33]: (maybe add a comment for future reference)

Comment (krfricke) [reply release/ray_release/buildkite/concurrency.py:33]: (this is needed if a test's resource usage cannot be determined)

Comment (mwtian) [reply release/ray_release/buildkite/concurrency.py:33]: Added a comment.

**Comment by mwtian** [comment]
Added a few assertions.

**Review by krfricke — APPROVED** [review]
Thanks!

**Pull request merged**